libcloud/dns/drivers/godaddy.py

```diff
@@ -0,0 +1,521 @@
+        id = '{0}:{1}'.format(name, type)
+        return Record(
+            id=id, name=name,
+            type=type, data=data,
+            zone=zone, driver=self,
+            extra=extra)
+
+    def update_record(self, record, name, type, data, extra=None):
+        """
+        Update an existing record.
+
+        :param record: Record to update.
+        :type  record: :class:`Record`
+
+        :param name: Record name without the domain name (e.g. www).
+                     Note: If you want to create a record for a base domain
+                     name, you should specify empty string ('') for this
+                     argument.
+        :type  name: ``str``
+
+        :param type: DNS record type (A, AAAA, ...).
+        :type  type: :class:`RecordType`
+
+        :param data: Data for the record (depends on the record type).
+        :type  data: ``str``
+
+        :param extra: (optional) Extra attributes (driver specific).
+        :type  extra: ``dict``
+
+        :rtype: :class:`Record`
+        """
+        if extra is None:
+            extra = {}
+        new_record = {}
+        if type is RecordType.SRV:
+            new_record = {
+                'type': type,
+                'name': name,
+                'data': data,
+                'priority': 1,
+                'ttl': extra['ttl'] if hasattr(extra, 'ttl') else 5,
+                'service': extra['service']
+                if hasattr(extra, 'service') else '',
+                'protocol': extra['protocol']
+                if hasattr(extra, 'protocol') else '',
+                'port': extra['port']
+                if hasattr(extra, 'port') else '',
+                'weight': extra['weight']
+                if hasattr(extra, 'weight') else ''
+            }
+        else:
+            new_record = {
+                'type': type,
+                'name': name,
+                'data': data,
+                'priority': 1,
+                'ttl': extra['ttl'] if hasattr(extra, 'ttl') else 5
+            }
+        self.connection.request(
+            '/v1/domains/{0}/records'.format(
+                record.zone.domain), method='PUT',
+            data=[new_record])
+        id = '{0}:{1}'.format(name, type)
+        return Record(
+            id=id, name=name,
+            type=type, data=data,
+            zone=record.zone, driver=self,
+            extra=extra)
+
+    def get_record(self, zone_id, record_id):
+        """
+        Return a Record instance.
+
+        :param zone_id: ID of the required zone
+        :type  zone_id: ``str``
+
+        :param record_id: ID of the required record
+        :type  record_id: ``str``
+
+        :rtype: :class:`Record`
+        """
+        parts = record_id.split(':')
+        result = self.connection.request(
+            '/v1/domains/{0}/records/{1}/{2}'.format(
+                zone_id,
+                parts[1],
+                parts[0])).object
+        if len(result) is 0:
```

Comment:
This would probably work most of the time, but it's kinda Python implementation specific so you should only use `is` to compare for True, False and None since `is` checks the reference and not the actual value.

See https://stackoverflow.com/questions/1504717/why-does-comparing-strings-in-python-using-either-or-is-sometimes-produce